tests/testsuite/directory.rs

```diff
@@ -784,3 +784,41 @@ Caused by:
         .with_status(101)
         .run();
 }
+
+#[cargo_test]
+fn custom_root_path() {
+    let p = ProjectBuilder::new(paths::root())
+        .no_manifest() // we are placing it in a different dir
+        .file(
+            "ws_root/Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.1.0"
+                edition = "2015"
+                authors = []
+            "#,
+        )
+        .file("ws_root/src/lib.rs", "invalid;")
+        .build();
+
+    // Crucially, the rustc error message below says `ws_root/...`, i.e.
+    // it is relative to our fake home, not to the workspace root.
+    p.cargo("check")
+        .arg("-Zroot-path=.")
+        .arg("--manifest-path=ws_root/Cargo.toml")
+        .masquerade_as_nightly_cargo(&["-Zroot-path"])
+        .with_status(101)
+        .with_stderr_data(str![[r#"
+[CHECKING] foo v0.1.0 ([ROOT]/ws_root)
+[ERROR] expected one of `!` or `::`, found `;`
+ --> ws_root/src/lib.rs:1:8
```

Comment:
Could you add this test in a previous commit, showing the output without `-Zroot-path=.` and passing?  Then the feature commit would only have the feature work, addition of `-Zroot-path=.`, and the test output update.  This makes it very clear how the output changes with this.

Reply:
I'll add that to my todo list, but by dev time is extremely limited so no promises when I can get around to fine-tune the git history here.